### PR TITLE
Added keyv 5 support for KeyvAdapter

### LINKED ISSUE #457: keyvAdapter: support Keyv 5
Keyv 5 was released today. <strike>It appears to have [some issues](https://github.com/jaredwray/keyv/issues/1113), but once resolved</strike> It would be nice to support the new version.

Someone tried it already and got this type error: https://github.com/jaredwray/keyv/issues/1113#issuecomment-2301217699

### PR DESCRIPTION
**Pull request by sroebert**
This implements support for keyv version 5, which fixes #457 and #459.

### CODE REVIEW

**Review by doanthai — COMMENTED** [review]
LGTM!

**Comment by cupofjoakim** [comment]
We'd also appreciate to go through, whenever there's time! :)

**Comment by Rikusor** [comment]
Here is also a team that is eagerly waiting this to go through!

File: packages/keyvAdapter/README.md [hunk 1]
```diff
@@ -18,9 +18,10 @@ In Apollo Server v3:
 import { ApolloServer } from "apollo-server";
```
Comment (glasser) [inline packages/keyvAdapter/README.md:18]: Could you add something to the README explaining that (if I understand correctly) v4 of this package works with keyv v5, and you can use v3 of this package for keyv v4?

Comment (sroebert) [reply packages/keyvAdapter/README.md:20]: Yes, I added a note about it.

**Pull request merged**